### PR TITLE
Fix for #58

### PR DESCRIPTION
**Pull request by kynx**
* XCache adapter now checks version > 3.1.0 when running under CLI
* fixed broken internalSetItem() with empty namespace

Don't got the chops to get travis building the xcache extension so the tests can run - doesn't look like it's avail from pecl (I get mine from brew). Anyone got any ideas?

### CODE REVIEW

**Comment by kynx** [comment]
Ach, since github doesn't link from the titles, this addresses #58

File: src/Storage/Adapter/XCache.php [hunk 1]
```diff
@@ -341,10 +341,17 @@ protected function internalSetItem(& $normalizedKey, & $value)
     {
         $options     = $this->getOptions();
         $namespace   = $options->getNamespace();
-        $prefix      = ($options === '') ? '' : $namespace . $options->getNamespaceSeparator();
+        $prefix      = ($namespace === '') ? '' : $namespace . $options->getNamespaceSeparator();
```
Comment (marc-mabe) [inline src/Storage/Adapter/XCache.php]: good catch :+1:

File: src/Storage/Adapter/XCache.php [hunk 2]
```diff
@@ -341,10 +341,17 @@ protected function internalSetItem(& $normalizedKey, & $value)
...
         $internalKey = $prefix . $normalizedKey;
         $ttl         = $options->getTtl();
 
+        if (is_object($value) || is_resource($value) || is_callable($value)) {
```
Comment (marc-mabe) [inline src/Storage/Adapter/XCache.php]: What does `is_callable` catch here for you as `is_callable` could be simple string, too.

**Comment by kynx** [comment]
Fixed in #65

**Pull request closed without merging**